earthpy/tests/test_clip.py

```diff
@@ -24,7 +23,7 @@
 """ Create Line Objects For Testing """
 
 linea = LineString([(1, 1), (2, 2), (3, 2), (5, 3)])
-lineb = LineString([(3, 4), (5, 7), (12, 2), (10, 5), (17, 17)])
+lineb = LineString([(3, 4), (5, 7), (12, 2), (10, 5), (9, 7.5)])
```

Comment:
here is the fix. if the  last set of points in lineb is outside of the clip extent then the entire line is NOT clipped.